google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1alpha2/StreamWriter.java

```diff
@@ -800,6 +839,18 @@ public void onResponse(AppendRowsResponse response) {
         if (response == null) {
           inflightBatch.onFailure(new IllegalStateException("Response is null"));
         }
+        if (response.hasUpdatedSchema()) {
+          if (streamWriter.getOnSchemaUpdateRunnable() != null) {
+            streamWriter.getOnSchemaUpdateRunnable().setUpdatedSchema(response.getUpdatedSchema());
+            try {
+              streamWriter.getOnSchemaUpdateRunnable().getStreamWriter().refreshAppend();
```

Comment:
Should it just be called on the current streamwriter?

Reply:
The execution of the Runnable and refresh append could race. I think we should simply run the runnable and let user handle everything.

Reply:
Sure, that would solve it.